### ISSUE TITLE
esp32 miner performance - do not use HA SHA1 calculation

### ISSUE THREAD

**Issue by pankleks** [duino-coin/duino-coin#940]
I think usage of hardware accelerated SHA1 is significantly decreasing possible chip performance.

Currently using ESP32 you can mine ~14-16kH on single so around 30 kH total.

On top of it hardware accelerated algorithm can't be used in parallel -> hence currently critical section is used to lock one hash calculation at given time.

I made test with software algorithm used in my STM32 miner - which can be run in parallel and in total I can get ~150 kH/s from the ESP32 chip.

In other words - current implementation of ESP32 is not using possible chip capabilities by far.

**Comment by revoxhere** [comment]
Closing due to inactivity/solved in the latest versions
